packages/babel-helpers/src/helpers.js

```diff
@@ -997,6 +1011,21 @@ helpers.iterableToArrayLimitLoose = helper("7.0.0-beta.0")`
   }
 `;
 
+helpers.unsupportedIterableToArray = helper("7.9.0")`
+  export default function _unsupportedIterableToArray(o) {
+    if (!o) return;
+    if (typeof o === "string") return Array.from(o);
+    var n = Object.prototype.toString.call(o).slice(8, -1);
+    if (n === "Object" && o.constructor) n = o.constructor.name;
```

Comment:
This should work with `es6-shim`. As an alternative, we could detect the presence of a [`_es6-shim iterator_`](https://github.com/paulmillr/es6-shim/blob/0d47be15894bb0f95068545aed69d388be8ce7d0/es6-shim.js#L281) property.

Reply:
By using Array.from, it already works with es6-shim without having to hardcode anything.

Reply:
`Array.from` is not supported on IE11. Can we use `arrayLikeToArray`?

Reply:
The problem is that Maps and Sets are not array-like.
I decided to use `Array.from` because because it is way more easily polyfilled than symbols.

WDYT about using `arrayLikeToArray` for strings/arguments/binary arrays, and `Array.from` for `Map`/`Set`?

Reply:
It's included with es6-shim tho, and imo it's completely reasonable to require anything faithfully polyfillable/shimmable, but symbols are not that.

@nicolo-ribaudo's suggestion works for me as well.